tests/Browser/LoginTest.php

```diff
@@ -0,0 +1,236 @@
+                    ->press('Login')
+                    ->assertPathIs('/')
+                    ->logout();
+        });
+    }
+
+    /**
+     * User logged in go to path '/admin'
+     * Case 2-2
+     */
+    public function testUserVisitAdmin()
+    {
+        $data = $this->getDatatTest();
+        $this->browse(function ($browser) use ($data) {
+            $browser->loginAs(User::find(2))
+                    ->visit($data['path']['admin'])
+                    ->assertPathIs($data['path']['user'])
+                    ->logout();
+        });
+    }
+
+    /**
+     * Admin login with email, password.
+     * Case 2-3
+     */
+    public function testAdminLogin()
+    {
+        $data = $this->getDatatTest();
+        $this->browse(function ($browser) use ($data) {
+            $browser->logout()
+                    ->visit($data['path']['login'])
+                    ->type('email', $data['email']['admin'])
+                    ->type('password', 'lifull@123')
+                    ->press('Login')
+                    ->assertPathIs($data['path']['admin']);
+        });
+    }
+
+    /**
+     * Admin logged go to path '/user'
+     * Case 2-4
+     */
+    public function testPathVisitUser()
+    {
+        $data = $this->getDatatTest();
+        $this->browse(function ($browser) use ($data) {
+            $browser->loginAs(User::find(1))
+                    ->visit($data['path']['user'])
+                    ->assertPathIs($data['path']['admin']);
+        });
+    }
+
+    /**
+     * User logout.
+     * Case 3-1
+     */
+    public function testUserLogout()
+    {
+        $this->browse(function ($browser) {
+            $browser->loginAs(User::find(2))
+                    ->logout()
+                    ->assertGuest();
+        });
+    }
+
+    /**
+     * Admin logout.
+     * Case 3-2
+     */
+    public function testAdminLogout()
+    {
+        $this->browse(function ($browser) {
+            $browser->loginAs(User::find(1))
+                    ->logout()
+                    ->assertGuest();
+        });
+    }
+
+    /**
+     * User logged, after remove user logined, reload browser:
+     * Case 3-3
+     */
+    public function testUserLoginAfterRemoveData()
+    {
+        $data = $this->getDatatTest();
+        // Create data test
+        $userDestroy = User::create([
+            'name'     => "user destroy",
+            'email'    => "usertest@lifull-tech.vn",
+            'role'     => \App\User::USER,
+            'password' => Hash::make("lifull@123")
+        ]);
+        // User login
+        $this->browse(function ($browser) use ($data, $userDestroy) {
+            $browser->loginAs($userDestroy)
+                    ->visit($data['path']['user'])
+                    ->assertPathIs($data['path']['user']);
+        });
+        // Remove user login in database
+        User::destroy($userDestroy->id);
+        // Reload brower
+        $this->browse(function ($browser) use ($data) {
```

Comment:
this action is not reload

Reply:
I fixed

Reply:
i can't use funciton script because it return array not use assert. i changed redirect path user